app/models/ect_at_school_period.rb

```diff
@@ -61,8 +48,17 @@ class ECTAtSchoolPeriod < ApplicationRecord
 
   # Instance methods
 
-  # lead_provider_name
-  delegate :name, to: :lead_provider, prefix: true, allow_nil: true
+  def latest_training_period
+    training_periods.latest_first.first
+  end
+
+  def lead_provider
+    latest_training_period&.lead_provider
+  end
+
+  def delivery_partner
+    latest_training_period&.delivery_partner
+  end
```

Comment:
We have a specific SO for the training stuff of an ECTAtSchoolPeriod.
https://github.com/DFE-Digital/register-early-career-teachers-public/blob/main/app/services/ect_at_school_periods/training.rb